scripts/testing/e2e-runner

```diff
@@ -0,0 +1,396 @@
+#!/bin/bash
+
+
+#
+# Default configuration, overridable on the command line.
+#
+# REMOTE_REPO:   remote repository to clone
+# CLONED_REPO:   local location to clone to
+# RESULT_ROOT:   directory for publishing results
+# RESULT_NAME:   subdirectory for publishing results of this test run
+# SKIP_WORKTREE: skip preparing worktree, assume it is already done
+# E2E_TESTS:     e2e tests to run
+
+REMOTE_REPO="${REMOTE_REPO:-https://github.com/containers/nri-plugins}"
+CLONED_REPO="${CLONED_REPO:-/opt/e2e-test/nri-plugins/nri-plugins}"
+TEST_BRANCH="${TEST_BRANCH:-main}"
+RESULT_ROOT="${RESULT_ROOT:-/opt/e2e-test/nri-plugins/results}"
+RESULT_NAME="${RESULT_NAME:-$(date +%Y-%m-%d-%H-%M)}"
+PROXIES_SRC=""
+SKIP_WORKTREE="${SKIP_WORKTREE:-}"
+E2E_TESTS="${E2E_TESTS:-}"
+
+
+# This script.
+SELF="$0"
+TEST_SUITE_DIR=policies.test-suite
+GIT_SHA1="git.sha1"
+GIT_DESC="git.describe"
+
+
+#
+# Derived configuration, set up after command line parsing
+#
+# RESULT_DIR: directory for publishing results of this test run
+# WORKTREE:   git worktree for this test run
+# LOG:        log file for this test run
+
+RESULT_DIR=""
+LOG=""
+WORKTREE=""
+
+usage() {
+    local exit_code="${exit_code:-0}"
+
+    echo "usage: ${1:-$SELF} [options] [<e2e-tests>]"
+    echo
+    echo "The possible options are:"
+    echo "  --source-proxies <file>  source proxy environment variables from file"
+    echo "  --remote|r <repo>        remote repository to clone/fetch from"
+    echo "  --branch|-b <branch>     name of the remote branch to test"
+    echo "  --local|-l <dir>         local directory to clone remote respository into"
+    echo "  --results|--out <dir>    top-level directory for publishing results"
+    echo "  --name|-n) <name>        subdirectory for results of this run"
+    echo "  --skip-worktree)         skip worktree preparation, assume already done"
+    echo "  --help)                  show this help"
+
+    exit $exit_code
+}
+
+info() {
+    echo 1>&2 "$@"
+}
+
+error() {
+    echo 1>&2 "error: $@"
+}
+
+fatal() {
+    echo 1>&2 "fatal error: $@"
+    exit 1
+}
+
+must-cd() {
+    if ! cd "$1"; then
+        fatal "failed to change directory to $1"
+    fi
+}
+
+must-mkdir() {
+    local dir="$1"; shift
+    if ! mkdir -p "$dir"; then
+        fatal "failed to create ${*:+$* }directory $dir"
+    fi
+}
+
+must-mv() {
+    if ! mv $*; then
+        fatal "mv $* failed"
+    fi
+}
+
+source_proxies() {
+    local var
+
+    if [ -z "$PROXIES_SRC" ]; then
+        info "no proxies to set up..."
+        return 0
+    fi
+
+    info "setting up proxies..."
+    if ! source "$PROXIES_SRC"; then
+        fatal "failed to source proxies from $src"
+    fi
+
+    for var in ${!http*} ${!HTTP*} ${!no_*} ${!NO_*}; do
+        case $var in
+            *_proxy|*_PROXY)
+                info "exporting proxy envrionment variable $var (with value ${!var})..."
+                export $var
+                ;;
+        esac
+    done
```

Comment:
This nice bash-fu just caught my untrained eye 😅 How about spelling out the exact variable names (and making it easier to read, for me at least). And ignoring variables like `no_PROXY` :)

```suggestion
    for var in http_proxy https_proxy no_proxy HTTP_PROXY HTTPS_PROXY NO_PROXY; do
        if [ -n "${!var}" ]; then
                info "exporting proxy envrionment variable $var (with value ${!var})..."
                export $var
        fi
    done
```

Of course it's not 100% the same as in my suggestion empty variables are not exported

...you can disregard this comment if you don't like it

Reply:
Yeah, probably better that way, since we know all the variables of interest. Updated.